sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,328 @@
+- Added initial implementation of the experimental SYCL Graph extension [`sycl_ext_oneapi_graph`](doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc)  [35465da4824a][f9d4830babca][72341ee375d6][96a605095665]
+- Added `dimensions` member to item/range-like types. [9c7bd9ea046f]
+- Added support of `queue::priority_*` properties for OpenCL backend. [25fd689999f7]
+- Implemented initial version of the SYCL Native CPU Plug-in designed in [SYCLNativeCPU](doc/design/SYCLNativeCPU.md). [b2d0837e9644] [5b501eeca963]
+- Added USM calls parameter verification layer to `sycl-trace` which provides improved diagnostic on accessing invalid device memory under USM. [758bd264bb90]
+- Added `__imf_max/min/hadd/fast_*` functions to imf device libraries. [fe01366e8b08][d96e5075303a]
+- Introduced and implemented new `sycl::ext::oneapi::experimental::info::device::architecture` device descriptor and `device::ext_oneapi_architecture_is(ext::oneapi::experimental::architecture)` host API as part of [`sycl_ext_oneapi_device_architecture`](doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc)  extension (for Level Zero and OpenCL). [d0b01b265d92]
+- Added experimental implementation of [`sycl_ext_intel_grf_size`](doc/extensions/experimental/sycl_ext_intel_grf_size.asciidoc) [370aa2a01711]
+- Added experimental implementation of [`sycl_ext_oneapi_bindless_images`](doc/extensions/experimental/sycl_ext_oneapi_bindless_images.asciidoc) [58a8f2001d8b]
+- Experimental [sycl_ext_oneapi_device_global](doc/extensions/experimental/sycl_ext_oneapi_device_global.asciidoc) extension is supported now. [d812d1e13ab5]
+- Enabled media API that works with image accessors for ESIMD. Those accessors do not depend on stateful/stateless mode. [b95b2d88e446]
+- Added complex literals support to `sycl::ext::oneapi::experimental::complex`. [83555cd511cc]
+- Allowed implicit conversion from `std::experimental::simd_mask` to `ESIMD::simd_mask`. [de5b479a2de9]
+- Added `host_task` image accessor support. [b6faefb08e23]
+- Started to support 64-bit offsets with accessors in stateless mode for ESIMD APIs. [f28ae00bf6a3][d04ebb03c1c8][62955fc20e14]
+- Added `__imf_llabs` to the imf device libraries. [3c19581f828c]
+- Added explicit conversion of `multi_ptr<T>` to `multi_ptr<const T>`. [10d2f5b613f3]
+- Added support for scalar logical operators with group algorithms. [11ac73003056][90406b2ba07c]
+- Started to print supported SG sizes in `sycl-ls --verbose`. [12515177ab03]
+- Implemented device query for 64 bit atomic support in ESIMD emulator. [c40baa6db64b]
+- Added support of 16 bit data for `lsc_atomic_update` and `lsc_slm_atomic_update` ESIMD API. [3028d82a75d2]
+- Added support for per-kernel auto GRF mode specification, and reimplemented the feature using kernel properties. Introduced a new property named `sycl::detail::register_alloc_mode` which takes in an enum `sycl::detail::register_alloc_mode_enum` with values: `automatic` and `large`. [f363bb272b5a]
```

Comment:
If I'm not mistaken, this is likely an intermediate step we had before finalizing GRF to `sycl_ext_intel_grf_size` extension. Tagging @sarnex to double-check me here, but most likely we only want to mention the final version, i.e. the extension

Reply:
You are right.
So, according to https://github.com/intel/llvm/commit/370aa2a01711 `sycl_ext_intel_grf_size` provides new API to set GRF mode (and it must be used from now on) and that commit makes old API `sycl::detail::register_alloc_mode` (which is described here) deprecated.

At the same time in this release window implementation of old API was changed with ABI breaking changes here - https://github.com/intel/llvm/commit/f363bb272b5a. Shortly, we changed old API with ABI break, but then implemented new API and deprecated old API. So, mentioning of this change is not needed at all.

Instead, I just added to ABI breaking section that we removed old `set_kernel_properties` API and `use_double_grf/use_large_grf` properties here https://github.com/intel/llvm/commit/f363bb272b5a in favor of new extension.